File: job-portal-system/services/job-portal-user-service/src/main/java/com/hope/job/service/AuthService.java
package com.hope.job.service;

import com.hope.job.payload.AuthResponse;
import com.hope.job.payload.LoginRequest;
import com.hope.job.payload.SignupRequest;

public interface AuthService {
    AuthResponse signup(SignupRequest req);

    AuthResponse login(LoginRequest req);
}

File: job-portal-frontend/src/lib/services/auth.service.ts
import { apiClient } from "@/lib/api/client";
import type { AuthResponse, LoginRequest, SignupRequest } from "@/types";

export const authService = {
  login: (data: LoginRequest) =>
    apiClient.post<AuthResponse>("/auth/login", data, { userId: null }),
  signup: (data: SignupRequest) =>
    apiClient.post<AuthResponse>("/auth/signup", data, { userId: null }),
};
